independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/BootstrapAppModelResolver.java

```diff
@@ -281,6 +291,31 @@ public boolean visitLeave(DependencyNode node) {
                 .build();
     }
 
+    private void collectPlatformProperties(AppModel.Builder appBuilder, List<Dependency> managedDeps)
+            throws AppModelResolverException {
+        for (Dependency d : managedDeps) {
+            // assuming platform properties artifacts have the same reserved groupId
+            Artifact artifact = d.getArtifact();
+            if (artifact.getArtifactId().endsWith(BootstrapConstants.PLATFORM_PROPERTIES_ARTIFACT_ID_SUFFIX)) {
+                final AppArtifact appArtifact = artifact.getExtension().equals("properties") ? toAppArtifact(artifact)
+                        : new AppArtifact(artifact.getGroupId(), artifact.getArtifactId(), artifact.getClassifier(),
```

Comment:
creating new AppArtifact looks a bit weird to me. When will it be necessary to create it? Are we sure that `resolve(appArtifact)` in this case returns an existing thing?

Reply:
It's just making sure the type of the artifact is always `properties`. If we go with this approach then `BootstrapConstants.PLATFORM_PROPERTIES_ARTIFACT_ID_SUFFIX` is becoming a reserved suffix requiring artifacts with this suffix to be of `properties` type. Because that's what we are looking for. OTOH, we could support other formats too.
You have a point. The reason I did it this way was I didn't add the `type` of the artifact to the BOM, so it appeared here as a `jar`, so I thought that could be a typical "mistake" and decided to not require `<type>properties</type>` in the BOM.